File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int dp[] = new int[2];
        dp[0] = 1;
        dp[1] = 1;
        for(int i = 2; i <= n; i++){
            dp[i % 2] = dp[(i - 1) % 2] + dp[(i -2) % 2];
        }
        return dp[n % 2];
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [1, 1]
        for i in range(2, n+1):
            dp[i % 2] = dp[(i - 1) % 2] + dp[(i - 2) %2]
        return dp[n % 2]